### ISSUE TITLE
Replace Bulma?

### ISSUE THREAD

**Issue by mikestefanello** [mikestefanello/pagoda#70]
[Bulma](https://bulma.io/) is a great project and I have nothing at all against it, but after coming across [Tabler](https://tabler.io/) recently, I am very excited to start using it and I think it would be a great fit for this project. It's exactly what I was looking for when this project started. From the looks of it, using nothing more than HTML, you can easily build a beautiful UI for practically any web app. I originally wanted a CSS kit that didn't have a JS dependency, but given the interactivity this provides out of the box, you may not even need Alpinejs (though I do plan to keep that included).

Are there any other good candidates out there?

**Comment by Jimmy99** [comment]
What changes would have to be made to accommodate Tabler? I would give it a try

**Comment by mikestefanello** [comment]
Thanks for offering to help. I actually spent a decent amount of time going down this path. It's quite a bit of work because you'll have to modify all of the HTML. It's also not a straight-forward conversion because Tabler has a ton of different components/layouts/etc to choose from, so there's some design choices to be made. I'm torn because of the Bootstrap and JS dependencies of Tabler, but then again, it does provide a lot right out of the box which is especially useful since I assume most people using this project are Go devs who don't want to spend much time dealing with tons of JS/CSS (if any at all). It also renders Alpine unnecessary for the most part, unless I decided to include something that Tabler couldn't do itself. I did like the idea of including it since it helps to highlight that you don't need React/Vue/Svelte to build out a good UI; which is one reason why so many go the route of using Go only for an API. And if you do include it, then you now have a lot of JS (htmx, bootstrap, tabler, alpine).

Then again, there's so many options out there that I'm curious how many people using this project would and do even keep whatever CSS default is in here. I'm not sure yet what I think is best for this project. Tabler is probably what I would personally prefer if I were using this to build out an app.

Maybe there's other options to consider before changes are made? I haven't looked too much and I know the frontend landscape changes rapidly. If you have any other suggestions, please share.

**Comment by gobijan** [comment]
Maybe for now just migrate to Bulma v1 wich is a complete rewrite with many refinements.

BeerCSS is also quite mighty and very easy to use.

**Comment by PraveenGandhi** [comment]
https://adminlte.io/ can be another option.

V4 (built with Bootsrrap 5) is WIP: https://github.com/ColorlibHQ/AdminLTE/releases/tag/v4.0.0-beta2

**Comment by mikestefanello** [comment]
Thanks for sharing.

I'm really not a fan of material design at all so I don't think I would want to go with BeerCSS.

I would say Tabler is vastly superior to AdminLTE.

**Comment by rlnorthcutt** [comment]
For what it's worth, I think it is better to go with fewer dependencies, which is why Bulma makes sense. However, perhaps there could be a network of forks that have different frameworks for the UI?

That way, if anyone wants to build/maintain a Tabler version, they can own that part and still inherit all the other great work here.

**Comment by mikestefanello** [comment]
I agree - and that was my original reasoning for choosing Bulma. Let's just keep it at that.

There's endless options out there and they're constantly changing, so I have to assume everyone is going to have a different preference. And even if you happen to like what's included here, you're not going to stick with the example templates so you're replacing those regardless. So, it really doesn't matter at all what's included.. this project just needs something simple that works.

I'll leave this open for now, until the port to v1 is done, and in case anyone has a suggestion for an alternative that should be considered.

**Comment by mikestefanello** [comment]
I'm going to close this since Tabler is off the table.

I recently upgraded to Bulma 1.x but I spent a lot of time (again) searching for viable alternatives. I surprisingly can't find any that I'm satisfied with yet or like better than Bulma so we'll stick with that for now. I thought PicoCSS was nice but way too limited out-of-the-box (which is by design, of course).

If any one has any other suggestions, please let me know.

**Comment by mikestefanello** [comment]
I'm re-opening this since I'm considering dropping Bulma again. This time, after a long investigation (and hesitation), I'm leaning towards [DaisyUI](https://daisyui.com/) which is a component library built on [Tailwind](https://tailwindcss.com/). I'd like to get any feedback anyone might have. I've stayed away from Tailwind, despite the huge popularity, community, resources, etc, because 1) I don't want to have to use npm, 2) I didn't want to deal with any extra build processes, systems, etc, 3) I dislike the endless wall of classes you need.

NPM is no longer needed with the standalone CLI. With air, we can simply extend the build command to compile the styles. DaisyUI provides very simple classes for components (but you still have access to any Tailwind utility class you might need).

I posted the WIP start to the code here: https://github.com/mikestefanello/pagoda/pull/111

**Comment by mikestefanello** [comment]
[Code](https://github.com/mikestefanello/pagoda/pull/111) is functionally complete but just needs review and cleanup - and the _readme_ needs a lot of updating. Feel free to test things out, if you're interested. Here's a look the styling out of the box. I included gomponent components for a handful of daisy components - as well as a starting icon library.

![Image](https://github.com/user-attachments/assets/db1b3255-379d-485f-bee7-f2dd01787907)

**Comment by mikestefanello** [comment]
@zoe-edwards Thanks for the thoughts and feedback. I usually stay far away from the frontend but I do agree with what you said. That's one of the reasons why I like and opted for DaisyUI alongside Tailwind. The aim of Pagoda is to make everything as easy as possible to get started and be productive with, and to provide simple patterns to follow - I think DaisyUI provides that.

In the end, whatever the default is probably doesn't matter too much, because I assume most people are just going to pick whatever they want, and throw away most of the frontend provided. So many people seem to prefer Tailwind, for good reason, so I do like the idea of including the setup for it out-of-the-box to make things easy.. if you want to switch back to some simple CDN option like Bulma, you can remove the Tailwind tooling in a few seconds. I was a bit concerned about including some the added complexity of downloading the CLI and re-compiling the CSS with air, etc, but it really is quite minimal. 

I've been watching and considered the new [BasecoatUI](https://basecoatui.com/) project which is a Shadcn clone but without React (and soon to be without Tailwind); but I think it's a bit too early and the Tailwind ecosystem and DaisyUI maturity is so hard to compete with.
